src/path_tracking_pid_local_planner.cpp

```diff
@@ -215,22 +215,22 @@ bool TrackingPidLocalPlanner::computeVelocityCommands(geometry_msgs::Twist & cmd
   if (dt.isZero()) {
     ROS_ERROR_THROTTLE(
       5, "dt=0 detected, skipping loop(s). Possible overloaded cpu or simulating too fast");
-    cmd_vel = geometry_msgs::Twist();
+    auto cmd_vel = geometry_msgs::Twist();
     cmd_vel.linear.x = pid_controller_.getControllerState().current_x_vel;
     cmd_vel.angular.z = pid_controller_.getControllerState().current_yaw_vel;
-    return true;  // False is no use: https://github.com/magazino/move_base_flex/issues/195
+    return cmd_vel;
   }
   if (dt < ros::Duration(0) || dt > ros::Duration(DT_MAX)) {
     ROS_ERROR("Invalid time increment: %f. Aborting", dt.toSec());
-    return false;
+    return std::nullopt;
   }
   try {
     ROS_DEBUG("map_frame: %s, base_link_frame: %s", map_frame_.c_str(), base_link_frame_.c_str());
     tfCurPoseStamped_ = tf_->lookupTransform(map_frame_, base_link_frame_, ros::Time(0));
   } catch (const tf2::TransformException & ex) {
     ROS_ERROR("Received an exception trying to transform: %s", ex.what());
     active_goal_ = false;
-    return false;
+    return std::nullopt;
```

Comment:
The MBF interface that we implement supports returning specific error codes for several of these failure modes. For example, and `TransformException` could return `uint8 TF_ERROR = 111` to indicate a transformation went wrong. This way the caller can figure out what went wrong.

I think `std::optional` doesn't really map well onto that. The command velocity is not optional. It is required, but it can go wrong due to various errors. I think raising exceptions for these errors would be better than the current implementation.

Reply:
Please be aware that I did not change the behavior of the code, I just refactored an output parameter. Based on the behavior of the previous implementation, `std::optional` is a perfect fit. The velocity command _is_ optional, because the operation can fail.

However, if you want to improve the code further and have this helper function also return the reason for the failure (using the error code you mentioned), I suggest we switch to a type from the Boost Outcome library, like `boost::checked<T, E>` which closely resembles the proposed `std::expected`. It is a type which represents a value of type T on success or an error value of type E on failure. But I think that should be done in a separate PR.

Reply:
Just to sure we're talking about the same thing here: the overload of `TrackingPidLocalPlanner::computeVelocityCommands()` I refactored in this PR, is no longer an interface function. We stopped inheriting from `nav_core::BaseLocalPlanner` in another PR. So now this is just a private helper function.

Reply:
I'm not a fan of this approach, but I won't block this PR for that. I think we should definitely focus on the error handling part more in the future. `std::expected` or `boost::checked` is a good tip